documentcloud/documents/processing/info_and_image/main.py

```diff
@@ -904,29 +907,53 @@ def extract_text_position(data, _context=None):
     page_text_pdf_field = redis_fields.page_text_pdf(doc_id)
 
     # Grab the PDF
+    errored = False
+    pdf = None
     if not in_memory:
         # If not using in-memory Redis PDFs, read the whole doc file into memory
         with storage.open(doc_path, "rb") as doc_file:
             contents = doc_file.read()
             mem_file = io.BytesIO(contents)
-        pdf = pdfplumber.open(mem_file)
+        try:
+            pdf = pdfplumber.open(mem_file)
+        except Exception as exc:  # pylint: disable=broad-except
+            logger.exception(
+                "[Opening pdfplumber doc_id %s failed]", doc_id, exc_info=exc
+            )
+            errored = True
 
     # Write all the text positions to file
 
     # Go through each page
     for page_number in page_numbers:
         if in_memory:
             # If in-memory, use the Redis overlay PDF
+            errored = False
             overlay_mem_file = io.BytesIO(
                 REDIS.hget(page_text_pdf_field, f"{page_number}")
             )
-            pdf = pdfplumber.open(overlay_mem_file)
+            try:
+                pdf = pdfplumber.open(overlay_mem_file)
+            except Exception as exc:  # pylint: disable=broad-except
+                logger.exception(
+                    "[Opening pdfplumber doc_id %s failed]", doc_id, exc_info=exc
+                )
+                errored = True
 
-        extract_text_position_for_page(
-            pdf, doc_id, slug, page_number, access, in_memory
-        )
+        if not errored:
```

Comment:
I think a `try` `else` block might be clearer than using the `errored` flag
https://stackoverflow.com/questions/855759/python-try-else